apps/data-extraction/src/extractors/airship-parts.extractor.ts

```diff
@@ -0,0 +1,58 @@
+import { AbstractExtractor } from '../abstract-extractor';
+import { map, switchMap, tap } from 'rxjs/operators';
+import { XivapiEndpoint } from '@xivapi/angular-client';
+
+export class AirshipPartsExtractor extends AbstractExtractor {
+
+  protected doExtract(): any {
+    const parts = {};
+
+    this.getAllPages(this.getSearchEndpointWithQuery({
+      indexes: XivapiEndpoint.Item,
+      columns: 'ID,AdditionalData',
+      filters: 'FilterGroup=28',
+    })).pipe(
+      map((page) => page.Results.map((result) => {
+        return {
+          additionalData: result.AdditionalData,
+          itemId: result.ID
+        };
+      })),
+      switchMap((itemResults) => {
+        return this.get(this.getResourceEndpointWithQuery(XivapiEndpoint.AirshipExplorationPart, {
+          ids: itemResults.map((r) => r.additionalData).join(','),
+          columns: 'ID,Slot,Rank,Components,Surveillance,Retrieval,Speed,Range,Favor,RepairMaterials'
+        }))
+          .pipe(
+            map((page) => page.Results),
+            tap((partResults) => {
+              partResults.forEach((part) => {
+                parts[part.ID] = {
+                  id: part.ID,
+                  // TODO: Waiting for XIVAPI update for the Slot field
+                  slot: Math.trunc(+part.ID / 7), // part.Slot,
```

Comment:
This part will be done next tuesday, you can already implement it as if it was here 👍